src/packagedcode/componentjs.py

```diff
@@ -0,0 +1,154 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import json
+from packagedcode import models
+from packageurl import PackageURL
+import yaml
+
+class ComponentJSONMetadataHandler(models.NonAssemblableDatafileHandler):
+    """
+    Handle component JSON metadata files for package analysis.
+    """
+    datasource_id = "component_json_metadata"
+    path_patterns = ("*component.json",)
+    default_package_type = "library"
+    description = "component JSON package metadata file"
+
+    @classmethod
+    def parse(cls, location, package_only=False):
+        """
+        Parse the JSON metadata file at `location` and yield PackageData.
+        """
+        with open(location, "r", encoding="utf-8") as f:
+            data = json.load(f)
+
+        name = data.get('name') or data.get('repo', '').split('/')[-1]
+        if not name:
+            return
+
+        namespace = None
+        if 'repo' in data and '/' in data['repo']:
+            namespace, name = data['repo'].split('/', 1)
+        
+        package_data = dict(
+            datasource_id=cls.datasource_id,
+            type=cls.default_package_type,
+            name=name,
+            namespace=namespace,
+            version=data.get('version'),
+            description=data.get('description', ''),
```

Comment:
```suggestion
            description=data.get('description'),
```
We want the defaults to be `None` (or whatever is defined at the model for this attribute) always if we don't have any value.